test/es-module/test-esm-tla-unfinished.mjs

```diff
@@ -158,4 +174,31 @@ describe('ESM: unsettled and rejected promises', { concurrency: !process.env.TES
     assert.strictEqual(stdout, '');
     assert.strictEqual(code, 13);
   });
+
+  describe('with exit listener', () => {
```

Comment:
I don't think it's specific to this PR but this pattern of appending tests that spawn child processes in the same file is really getting out of hand...I think at some point we'll need to start enforcing "if it can be done in a new file, do it in a new file" before test failures become completely undebuggable...

Reply:
Opened a separate doc update PR: https://github.com/nodejs/node/pull/57028